sdk/appcenter-crashes/src/main/java/com/microsoft/appcenter/crashes/Crashes.java

```diff
@@ -371,12 +422,30 @@ protected synchronized void applyEnabledState(boolean enabled) {
             /* Delete cache and in memory last session report. */
             mErrorReportCache.clear();
             mLastSessionErrorReport = null;
+            mContext.unregisterComponentCallbacks(mMemoryWarningListener);
+            SharedPreferencesManager.remove(PREF_KEY_MEMORY_RUNNING_LEVEL);
         }
     }
 
     @Override
     public synchronized void onStarted(@NonNull Context context, @NonNull Channel channel, String appSecret, String transmissionTargetToken, boolean startedFromApp) {
         mContext = context;
+        mMemoryWarningListener = new ComponentCallbacks2() {
```

Comment:
we should create the object in applyEnabledState(true) only and nullify when false after unregistering